### ISSUE TITLE
Controlled Threading

### ISSUE THREAD

**Issue by codingo** [codingo/Reconnoitre#71]
Currently for each machine returned Reconnoitre will start a new thread for the remote hosts. This means that if 255 machines are available, 255 instances of NMap will fire in the background.

A change to the way threads are started and managed in the application and to also allow them to be controlled by the command line would resolve this (no small undertaking).

**Comment by codingo** [comment]
I'll get to this eventually, but if anybody wishes to try ahead of time the code required could be lifted from https://github.com/codingo/Interlace

**Comment by hrishikeshSuresh** [comment]
I would like to work on this if it hasn't been done already.

**Comment by codingo** [comment]
@hrishikeshSuresh yes, feel free. If you can put a paypal e-mail in the pull request so I can send a token of thanks as well.

**Comment by hrishikeshSuresh** [comment]
I went through Interface and I think I have to add a few arguments to `reconnoitre` command which will be used to manage the threads. So what are the basic things I need to cover? I'm expecting few arguments to control maximum number of threads, ip addresses of machines that will run the nmap? Anything else I have missed?
